File: Remove Duplicates From Sorted LinkedList/LinkedList.cpp
//
// Created by MRIGANKA on 3/11/2022.
//

#include <iostream>
#include "LinkedList.h"

LinkedList::LinkedList() : head(nullptr) {}

void LinkedList::insert(int data) {
    if (head == nullptr) {
        head = new Node(data);
    } else {
        Node *temp = head;
        while (temp->next != nullptr) {
            temp = temp->next;
        }
        temp->next = new Node(data);
    }
}

void LinkedList::remove_duplicates() {
    Node *temp = head;
    while (temp->next) {
        if (temp->data == temp->next->data) {
            temp->next = temp->next->next;
        } else {
            temp = temp->next;
        }
    }
}

void LinkedList::print() {
    Node *temp = head;
    while (temp) {
        std::cout << temp->data << std::endl;
        temp = temp->next;
    }
}

File: Add one to the linked list/src/com/Releationship/Linkedlist.java
package com.Releationship;

public class Linkedlist {
    public Node head;

    public Linkedlist() {
        this.head = null;
    }

    public void insert(int data) {
        if(head == null) {
            head = new Node(data);
        }
        else {
            Node temp = head;
            while(temp.next != null) {
                temp = temp.next;
            }
            temp.next = new Node(data);
        }
    }

    private Node reverse(Node head) {
        Node current = head, prev = null, next = null;
        while(current != null) {
            next = current.next;
            current.next = prev;
            prev = current;
            current = next;
        }
        head = prev;
        return head;
    }

    public void addOne() {
        Node temp = reverse(head);
        Node temp1 = temp;
        int sum = 0, carry = 1;
        while(temp != null) {
            sum = (temp.data + 1) % 10;
            carry = (temp.data + 1) / 10;
            temp.data = sum;
            temp = temp.next;
        }
        head = reverse(temp1);
    }

    public void print() {
        Node temp = head;
        while(temp != null) {
            System.out.println(temp.data);
            temp = temp.next;
        }
    }
}